### ISSUE TITLE
What is the basic difference between general model and matting model?

### ISSUE THREAD

**Issue by skedin** [ZhengPeng7/BiRefNet#190]
Sorry that this should be a question, not an issue.
I'd like to know what is the basic difference of the general model (like HR_epoch_230 or epoch_244) and matting one (like HR-matting-epoch_135 or matting-epoch_100)?

The "matting results" I saw from other SOD/DIS projects are pretty much like that the prediction alpha (mask) is transparent as  we can see through the glass bottle, plastic bag or lamp bulbs. The objects have "transparency" so the prediction alpha values are also between 0~255.

By your training dataset and actual prediction mask, I can not clearly tell the difference of general one and matting one!
Could you please kindly give the hints?

Regards
Stephen

**Comment by ZhengPeng7** [comment]
Hi, Stephen; thanks for the interest, and plz don't care about whether it's a question or an issue.

It's easy to explain -- they differ from two main parts, i.e., the training sets and loss settings. The matting ones only had matting data (P3M-10k, AIM-2k, etc.), and no classification loss (BCE / IoU) was used there.

Actually, they did have obvious differences in the results of objects with transparencies. For example, you can see [my tweet](https://x.com/zhengpeng0108/status/1891776875938316644) of BiRefNet_HR-matting comparison with previous BiRefNet models and use [my online demo](https://huggingface.co/spaces/ZhengPeng7/BiRefNet_demo) to make a comparison on more examples.

**Comment by skedin** [comment]
The picture in X explains everything, thanks.

And since you mentioned that the matting model does not use BCE and IoU, by my reading of the code, I guess it should be in config.py:

# Loss
        if self.task in ['Matting']:
            self.lambdas_pix_last = {
                'bce': 30 * 1,
                'iou': 0.5 * 0,
                'iou_patch': 0.5 * 0,
                'mae': 100 * 1,
                'mse': 30 * 0,
                'triplet': 3 * 0,
                'reg': 100 * 0,
                'ssim': 10 * 1,
                'cnt': 5 * 0,
                'structure': 5 * 0,
            }
        elif self.task in ['General', 'General-2K']:
            self.lambdas_pix_last = {
                'bce': 30 * 1,
                'iou': 0.5 * 1,
                'iou_patch': 0.5 * 0,
                'mae': 100 * 1,
                'mse': 30 * 0,
                'triplet': 3 * 0,
                'reg': 100 * 0,
                'ssim': 10 * 1,
                'cnt': 5 * 0,
                'structure': 5 * 0,
            }
        else:
            self.lambdas_pix_last = {
                # not 0 means opening this loss
                # original rate -- 1 : 30 : 1.5 : 0.2, bce x 30
                'bce': 30 * 1,          # high performance
                'iou': 0.5 * 1,         # 0 / 255
                'iou_patch': 0.5 * 0,   # 0 / 255, win_size = (64, 64)
                'mae': 30 * 0,
                'mse': 30 * 0,         # can smooth the saliency map
                'triplet': 3 * 0,
                'reg': 100 * 0,
                'ssim': 10 * 1,          # help contours,
                'cnt': 5 * 0,          # help contours
                'structure': 5 * 0,    # structure loss from codes of MVANet. A little improvement on DIS-TE[1,2,3], a bit more decrease on DIS-TE4.
            }


I can see iou is opened for "general", closed for "matting", but why bce is opened for "matting"? (mutiplied by 0 means disabling it?)


Regards

**Comment by ZhengPeng7** [comment]
BCE theoretically has some bad effects on matting training. But it's not that kind of direct / thresholding like IoU. On the other hand, it can accelerate convergence and is important for non-matting data. So, I just kept that for all tasks.
